src/hotspot/cpu/x86/x86.ad

```diff
@@ -2633,6 +2633,23 @@ bool Matcher::supports_vector_calling_convention(void) {
   return EnableVectorSupport;
 }
 
+bool Matcher::should_attempt_register_biasing(const MachNode* mdef, int oper_index) {
+  if (mdef == nullptr) {
+    return false;
+  }
+
+  if (mdef->num_opnds() <= oper_index || mdef->operand_index(oper_index) < 0) {
+    return false;
+  }
+
+  // Returns true for MachNode corresponding to Intel APX NDD selection patterns which
+  // can be demoted to REX/REX2 encodings. For commutative operations with register
+  // operands, allocation of definition operand is biased towards both the operands.
+  return (((mdef->flags() & Node::PD::Flag_ndd_demotable) != 0) &&
```

Comment:
It is called either with ` oper_index == 1` or ` oper_index == 2`. Can you make it explicit that any other operand doesn't participate in register biasing? Also, I'd expand the check, so it becomes clear that 1st operand requires `Flag_ndd_demotable` and 2nd requires `Flag_ndd_demotable` + `Flag_ndd_commutative` set.